src/views/Profile/AdminTable/AdminDatePicker.tsx

```diff
@@ -32,18 +35,93 @@ interface IMaterialsDate {
   setChanges: (...params: any[]) => void;
 }
 
+type ISelectedDayType = {
+  bgColor: 'transparent' | 'grey';
+  value: string | '';
+};
+
+const datePickerReducer = (
+  state: {
+    firstSelectedDay: ISelectedDayType;
+    secondSelectedDay: ISelectedDayType;
+  },
+  action: { type: string; payload: ISelectedDayType },
```

Comment:
`type` is better to be typed as [a discriminated union](https://www.typescriptlang.org/docs/handbook/2/narrowing.html#discriminated-unions)

Reply:
Sorry, I didn't understand you before, but now I changed it.